core/services/gateway/handlers/vault/handler.go

```
@@ -275,34 +300,45 @@ func (h *handler) HandleNodeMessage(ctx context.Context, resp *jsonrpc.Response[
 		RawResponse: rawResponse,
 		ErrorCode:   errorCode,
 	}
-	return h.sendResponse(ctx, ur, successResp)
+	return h.sendResponse(ctx, ar, successResp)
 }
 
-func (h *handler) handleSecretsCreate(ctx context.Context, ar activeRequest) error {
+func (h *handler) handleSecretsCreate(ctx context.Context, ar *activeRequest) error {
 	l := logger.With(h.lggr, "method", ar.req.Method, "requestID", ar.req.ID)
-	var createSecretsRequests vaultcommon.CreateSecretsRequest
-	if err := json.Unmarshal(*ar.req.Params, &createSecretsRequests); err != nil {
+
+	createSecretsRequest := &vaultcommon.CreateSecretsRequest{}
+	if err := json.Unmarshal(*ar.req.Params, &createSecretsRequest); err != nil {
 		return h.sendResponse(ctx, ar, h.errorResponse(ar.req, api.UserMessageParseError, err))
 	}
-	if createSecretsRequests.RequestId == "" {
+
+	createSecretsRequest.RequestId = ar.req.ID
```

I'm beginning to believe we shouldn't have 2 places to put reqId for the user.
Here, if user puts different ids on json message vs here, we are silently ignoring this inner id without validating.

Instead we should harden our spec so that id can be provided by user only in 1 place, so no place for any confusion.

Just wondering out loud for later, nothing to change in this PR.

---

I agree 👍 I think the answer could be to split out the user-facing spec from the internal spec, with the difference being that the user-facing spec is the same as the proto message but without the requestID (since the ID will be provided by the user in the standard jsonrpc field)

Wdyt?

---

Yes. I would also want to remove the Owner field from user facing spec.
Because owner is generated during our auth process, and there's no point letting the user provide it. We will anyways not trust the owner fied that user provides.